src/packidx.rs

```diff
@@ -298,6 +298,8 @@ impl<'a> From<FileEntryRef<'a>> for FileEntry {
 pub struct ObjectMetadata {
     pub offset: u64,
     pub size: u64,
+    #[serde(default)] // New in Version::V2, None if missing.
```

Comment:
Note, I expect may have a version enum at some point which is why I have this here, but the enum turns out to be unnecessary to implement it for the moment.

One thing I did entertain was storing the modes in a separate array from the ObjectMetadata array. They're likely to mostly have the same value so will compress quite well. That said, it's an extra 4 bytes per unique file in the metadata so the simpler implementation is preferable for now. We'll still have the option of changing the format in a compatible way in the future.